<?xml version="1.0"?>
<launch>
	<!-- Include the RPLidar Driver -->
	<include file="$(find rplidar_ros)/launch/rplidar.launch" >
		<arg name="arg_serial_port" value="/dev/ttyUSB2"/>
		<arg name="arg_serial_baudrate" value="115200"/>
	</include>

	<!-- Rviz for visialization -->
	
	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find uav_planning)/rviz/rplidar_env.rviz" />

	<!-- UAV connection -->
	<!-- <node pkg="uav_planning" type="conn_uav.py" name="test" required="true" output="screen" >
		<param name="uav_conn_protocol" type = "string" value = "uart"/>
		<param name="uav_conn_baundrate" type = "int" value = "115200"/>
		<param name="uav_conn_serial_port" type = "string" value = "/dev/ttyACM0"/>
	</node> -->

	<!-- GCS connection -->
	<node pkg="uav_planning" type="conn_gcs.py" name="gcs_connector" required="true" output="screen" >
		<param name="gcs_conn_protocol" type = "string" value = "uart"/>
		<param name="gcs_conn_UDP_IP" type = "string" value = "192.168.1.120"/>
		<param name="gcs_conn_UDP_port" type = "string" value = "5555"/>
		<param name="gcs_conn_baundrate" type = "int" value = "115200"/>
		<param name="gcs_conn_serial_port" type = "string" value = "/dev/ttyUSB1"/>
	</node>
	<!-- Our node to control the drone -->
	<node pkg="uav_planning" type="task_achievement.py" name="uav_planner" required="true" output="screen">
		<param name="uav_conn_protocol" type = "string" value = "uart"/>
		<param name="uav_conn_baundrate" type = "int" value = "500000"/>
		<param name="uav_conn_serial_port" type = "string" value = "/dev/ttyUSB0"/>
	</node>
</launch>

